src/vos/vos_ts.h

```diff
@@ -328,8 +328,12 @@ vos_ts_alloc(struct vos_ts_set *ts_set, uint32_t *idx, uint64_t hash)
 		info = &ts_table->tt_type_info[0];
 	} else {
 		info = parent->te_info;
-		/* Allocated entry must have a real parent */
-		D_ASSERT((info->ti_type & 1) == 0);
+		if (info->ti_type & 1) {
+			/** this can happen if it's a duplicate key.  Return
+			 * NULL in this case
```

Comment:
Why not support overwrite if found duplicated akey? That may be normal case if someone update the same akey multiple times within the same distributed transaction.

Reply:
Will that be in a single vos call?  At least in current API, there is only one epoch at play here so the behavior is undefined.   We may have to rethink this some but I’d rather fix the immediate issue before deciding how to tackle the more generic problem.

Reply:
The DAOS API (daos_obj_update) does not forbid to update the same akey via single API call, right? For distributed transaction, more general cases are that the application may update the same akey multiple times via multiple vos calls within the same distributed transaction.

Reply:
Perhaps we'd clarify it in the manual of daos_obj_update(). I didn't see the value of updating same akey in single daos_obj_update() so far, do you have any use cases?

Reply:
This will have no effect on multiple vos calls case.   It's only an issue when two updates happen on same akey in same vos call with same epoch.   According to Mohamad, this has undefined behavior so I've provided a workable "do not crash" behavior.